projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-abstract-token-subscription-service.php

```diff
@@ -444,7 +444,12 @@ private function set_token_cookie( $token ) {
 		}
 
 		if ( ! empty( $token ) && false === headers_sent() ) {
-			setcookie( self::JWT_AUTH_TOKEN_COOKIE_NAME, $token, 0, '/', COOKIE_DOMAIN, is_ssl(), true ); // httponly -- used by visitor_can_view_content() within the PHP context.
+			$cookie_domain = COOKIE_DOMAIN;
+			if ( empty( $cookie_domain ) && ! empty( $_SERVER['HTTP_HOST'] ) ) {
+				$cookie_domain = '.' . wp_unslash( $_SERVER['HTTP_HOST'] ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash,WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			}
+
+			setcookie( self::JWT_AUTH_TOKEN_COOKIE_NAME, $token, 0, '/', $cookie_domain, is_ssl(), false ); // phpcs:ignore Jetpack.Functions.SetCookie.FoundNonHTTPOnlyFalse
```

Comment:
Set the cookie domain when 'COOKIE_DOMAIN' is empty and don't make it HTTP ONLY.

Reply:
Ok, so adding a leading dot to the domain is probably wrong because atomic sites don't have it.

It seems we have a problem here because the leading dot is always added:
https://github.com/Automattic/jetpack/blob/3cf4ae640be8cf9cb53f37e8445bed7b62519a53/projects/plugins/jetpack/extensions/shared/memberships.js#L112C14-L127